hardware/EcoDevices.h

```diff
@@ -19,14 +19,96 @@ class CEcoDevices : public CDomoticzHardwareBase
 		bool m_stoprequested;
 		boost::shared_ptr<boost::thread> m_thread;
 
-		P1Power m_p1power1, m_p1power2;
-		P1Gas   m_p1water, m_p1gas;
-		time_t m_lastSendData;
-		unsigned long m_lastusage1, m_lastusage2, m_lastwaterusage, m_lastgasusage;
+		uint32_t m_indexC1, m_indexC2, m_previousC1, m_previousC2;
+		time_t m_lastSendDataT1, m_lastSendDataT2, m_lastSendDataC1, m_lastSendDataC2;
+
+		typedef struct _tTeleinfo
+		{
+			uint8_t len;
+			uint8_t type;
+			uint8_t subtype;
+			int32_t ID;
+			std::string PTEC;
+			uint32_t PAPP;
+			uint32_t BASE;
+			uint32_t HCHC;
+			uint32_t HCHP;
+			uint32_t PEJP;
+			uint32_t EJPHN;
+			uint32_t EJPHPM;
+			uint32_t BBRHPJB;
+			uint32_t BBRHCJB;
+			uint32_t BBRHPJW;
+			uint32_t BBRHCJW;
+			uint32_t BBRHPJR;
+			uint32_t BBRHCJR;
+			std::string DEMAIN;
+			uint32_t previous;
+			std::string rate;
+			std::string tariff;
+			std::string color;
+			_tTeleinfo()
+			{
+				len = sizeof(_tTeleinfo) - 1;
+				type = 0;		 //pTypeP1Power;
+				subtype = 0;	 //sTypeP1Power;
+				ID=0;
+				PTEC="";
+				PAPP=0;
+				BASE=0;
+				HCHC=0;
+				HCHP=0;
+				PEJP=0;
+				EJPHN=0;
+				EJPHPM=0;
+				BBRHPJB=0;
+				BBRHCJB=0;
+				BBRHPJW=0;
+				BBRHCJW=0;
+				BBRHPJR=0;
+				BBRHCJR=0;
+				DEMAIN="";
+				previous=0;
+				rate="";
+				tariff="";
+				color="";
+			}
+		} Teleinfo;
+
+		Teleinfo m_teleinfo1, m_teleinfo2;
 
 		void Init();
 		bool StartHardware();
 		bool StopHardware();
 		void Do_Work();
 		void GetMeterDetails();
 };
+
```

Comment:
Translation done.

Also tried to fix include issues that may be the cause of tests failures while it compiles fine on my Raspberry. Now waiting for tests to start